qiskit_nature/circuit/library/ansatzes/succ_full.py

```diff
@@ -0,0 +1,208 @@
+        # Note, that we use `product` here, in order to also get those double
+        # excitations form an alpha excitation and a beta excitation. We will need those in the
+        # following post-processing step.
+        pool = itertools.product(alpha_excitations, beta_excitations)
+
+        for exc in pool:
+            # find the two excitations (Note: SUCCD only works for double excitations!)
+            alpha_exc, second_exc = exc[0], exc[1]
+            # shift the second excitation into the beta-spin orbital index range
+            beta_exc = (
+                second_exc[0],
+                second_exc[1],
+            )
+            # add the excitation tuple
+            occ: Tuple[int, ...]
+            unocc: Tuple[int, ...]
+            occ, unocc = zip(alpha_exc, beta_exc)
+            exc_tuple = (occ, unocc)
+            excitations.append(exc_tuple)
+            logger.debug("Added the excitation: %s", exc_tuple)
+
+        return excitations
+
+    def _validate_num_particles(self, num_particles):
+        try:
+            assert num_particles[0] == num_particles[1]
+        except AssertionError as exc:
+            raise QiskitNatureError(
+                "The SUCC_full Ansatz only works for singlet-spin systems. However, you specified "
+                "differing numbers of alpha and beta electrons:",
+                str(num_particles),
+            ) from exc
+
+    def _build_fermionic_excitation_ops(self, excitations: Sequence) -> List[FermionicOp]:
+        """Builds all possible excitation operators with the given number of excitations for the
+        specified number of particles distributed in the number of orbitals.
+        Args:
+            excitations: the list of excitations.
+        Returns:
+            The list of excitation operators in the second quantized formalism.
+        """
+        operators = []
+        excitations_dictionary: Dict[int, List[Tuple[Tuple[int, ...], Tuple[int, ...]]]] = {}
+        for exc in excitations:
+            exc_level = sum(exc[1])
+            if exc_level in excitations_dictionary:
+                excitations_dictionary[exc_level].append(exc)
+            else:
+                excitations_dictionary[exc_level] = [exc]
+        for exc_level, exc_level_items in excitations_dictionary.items():
+            final_op = FermionicOp.zero(register_length=self.num_spin_orbitals)
+            for exc in exc_level_items:
+                label = ["I"] * self.num_spin_orbitals
+                for occ in exc[0]:
+                    label[occ] = "+"
+                for unocc in exc[1]:
+                    label[unocc] = "-"
+                op = FermionicOp("".join(label), display_format="dense")
+                op -= op.adjoint()
+                # we need to account for an additional imaginary phase in the exponent (see also
+                # `PauliTrotterEvolution.convert`)
+                op *= 1j  # type: ignore
+                final_op += op
```

Comment:
The other key difference compared to the normal `UCC` is now that you sum the excitations of identical levels into one operator, resulting in them being associated to a single parameter in the parameterized ansatz. A very nice solution indeed :+1: 

I was actually hoping, that we could make this `"full"` variant of `SUCC` a keyword option of the `SUCCD` class, since these are quite related in that sense. This would remove the need for largely duplicating the rest of this class. Do you think you could update this PR to attempt such a change instead?

Reply:
Thank you very much for the advices! I just want to make sure, I am going to add one more input in SUCCD named "full",  if "full=False", it gives SUCCD, if "full=True", it gives SUCC_full. Am I right?

Reply:
Yes, in principal this is correct. I will think about whether we can find a better name than `full`, but for now, please go ahead and implement it like this :+1: